sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -556,7 +556,10 @@ public void flush(long timeoutMillis) {
                   e);
         } finally {
           if (item != null) {
-            item.getScope().clearTransaction();
+            final Scope scope = item.getScope();
+            if (scope.getTransaction() == transaction) {
+              scope.clearTransaction();
```

Comment:
On Android where the scope is shared this would race since in between another transaction could have been set.

Another approach would be:

`scope.clear(transaction)`
And internal conditional atomic reference swap (i.e: https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.compareexchange?view=net-5.0)